### DISCUSSION TITLE
How to make a rigid body float on water from t = 0.

### DISCUSSION THREAD

**Discussion by fluidkocy** [InteractiveComputerGraphics/SPlisHSPlasH#264] [Q&A] [answered]
Simulation of a rigid body floating on water. I want to create a situation where the rigid body is allowed to float above the water at a certain velocity until a certain time, and after that time, the rigid body is allowed to drift above the water.

Any suggestions would be appreciated.

**Comment by janbender** [comment]
I don't get the point. What exactly do you want to do? A dynamics rigid body always swims on top as long as its density is not too high. So rigid body with density of 500 would swim on water with density of 1000. And as a dynamic body you could also give it an initial velocity.

**Reply by fluidkocy** [reply]
How to make a rigid body float on water from t = 0.

Sorry for my lack of words.
As the title says, I want to know how to make a rigid body float on water from t = 0.
I want to create a situation where the rigid body is already floating on the water at t = 0 and the water is already at rest.

With my current method, the rigid body is dropped from above the water surface, so the water and the rigid body will not come to rest until some time has passed.

Also, the impact of the rigid body falling into the water will cause the rigid body to be out of position from where I want it to float in the water. Therefore, I would like to know how to create a situation where the rigid body is already floating in water from t=0.

If you have a better solution, please help me.

**Comment by janbender** [comment, marked as answer]
Ah, okay. Start the simulation where the body is dropped on the fluid surface. Export particle data (bgeo) and rigid bodies (obj). When everything comes to rest, stop the simulation. Take the last exported bgeo and obj file and set up a new scene. 

The fluid part should look like this:
```
"FluidModels": [
		{
			"particleFile": "lastframe.bgeo",
			"translation": [-2.0, 0.1, 0.0],
			"rotationAxis": [0, 1, 0],
			"rotationAngle": 0,
			"scale": [1, 1, 1]
		}
	]
```

Add a rigid body using the last obj file and give it an inital velocity or animate it (see example AnimatedBody_2D.json). 

Hope that helps.

**Reply by fluidkocy** [reply]
thank you very much !!
